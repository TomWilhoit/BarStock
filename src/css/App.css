/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE: 
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  /* background: #1d2023; */
}

body {
  height: 100%;
}

/* #endRegion Global */



.Header-container {
  padding-bottom: 2em;
}

.Footer-container {
  position: relative;
  width: 100%;
  text-align: center;
  /* background: #1d2023; */
}

