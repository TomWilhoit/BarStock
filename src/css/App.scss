/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE:
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  background: url(./dust_scratches.png);
}

body {
  height: 100%;
}

/* #endRegion Global */

#root {
  height: 100%;
}

.App {
  height: 100%;
}

.All-Content {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}


.Header-container {
  margin-bottom: 2em;
}


/* #region Content Container  */


.content-container {
  display: table;
  max-width: 1200px;
  width: 100%;
  margin: auto;
}

.Inventory-container {
  display: table-cell;
  width: 60%;
  padding: 0em 0em 0em 2em;
}

.Totals-container {
  display: table-cell;
  width: 40%;
  padding: 0em 2em 0em 2em;
  vertical-align: bottom;
}

/* #endRegion Content Container */


.Submit-order {
  button {
    width: 100%;
    border-radius: 4px;
    outline: none;
    background: #EEAD0E;
    color: #fff;
    text-transform: uppercase;
    font-weight: 100;
    font-size: 1em;
    padding: .2em;
    font-family: 'Abel', sans-serif;
    cursor: pointer;
    &.disabled {
      pointer-events: none;
      opacity: .2;
    }
    &.enabled {
      pointer-events: all;
      opacity: 1;
    }
  }
}


.Footer-container {
  width: 100%;
  text-align: center;
  margin-top: 2em;
}



@media only screen and (max-width: 800px) {
  .content-container {
    display: block;
  }
  .Inventory-container {
    margin: auto;
    display: block;
    width: 100%;
    max-width: 600px;
    padding: 0;
    padding-bottom: 2em;
  }
  .Totals-container {
    display: block;
    margin: auto;
    width: 100%;
    padding: 0;
    max-width: 600px;
  }
}
