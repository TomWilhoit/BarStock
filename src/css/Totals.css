.Totals {
    margin: auto;
    background: #cccccc;
    width: 100%;
    max-width: 600px;
    text-align: center;
    border:#231f20 solid 1px;
    padding: 1em 0;
    margin-top: 2em;
    margin-bottom: 2em;
}