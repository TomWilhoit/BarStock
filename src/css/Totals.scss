/********************
FONTS
font-family: 'Open Sans', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Raleway', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

*********************/

.Totals {
  background: #fff;
  box-shadow: 0px 1px 7px 0px rgba(35, 31, 32, .2);
  width: 100%;
  display: flex;
  flex-direction: column;
  margin: 0;
}

.total-label {
  text-align: center;
  font-family: 'Abel', sans-serif;
  background: #dadada;
}

.total-label h3 {
  color: #2c4969;
  margin: 0;
  padding: .2em;

  font-weight: 100;
}

/* #region Cart Items */
.cart-display {
  overflow-y: scroll;
  max-height: 250px;
}

.cart-products {
  display: flex;
  align-items: center;
  justify-content: space-between;
  align-items: center;
  padding: .5em .75em;
  border-bottom: solid 1px #ececec;
}
.cart-products:nth-of-type(2n) {
  background:#f7f7f7cc;
}

.item-product {
  width: 55%;
}
.item-product h5, .item-price h5 {
  font-family: 'Lato', sans-serif;
  margin: 0;
  font-size: .8em;
  font-weight: 100;
  color: #777777;
}

.item-price {
  width: 25%;
}

.item-delete {
  line-height: 0;
}
.item-delete button {
  padding: 0;
  border: none;
  background: none;
  color: #0098de;
  cursor: pointer;
  transition: all .3s ease;
  outline:none;
}
.item-delete button:hover {
  color:#dadada;
}


/* #endregion Cart Items */


/* #region Totals Sum */

.cost-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  text-align: center;
  background: #ececec;
}

.total-cost {
  padding: .5em 1em;
  background: #2c4969;
  width: 50%;

  h2 {
   margin: 0;
   font-family: 'Abel', sans-serif;
   color:#fff;
   font-size: 1.3em;
 }

 h3 {
   margin: 0;
   font-family: 'Abel', sans-serif;
   color:#fff;
   font-size: 1.3em;
   letter-spacing: .01em;
 }
}

.returns-cost {
  padding: .5em 1em;
  width: 50%;

  h2 {
    margin: 0;
    font-weight: 100;
    font-size: .8em;
    color:#2c4969;
    font-family: 'Abel', sans-serif;
  }

  h3 {
    margin: 0;
    font-weight: 100;
    color:#2c4969;
    font-family: 'Abel', sans-serif;
  }
}

/* #endregion Totals Sum */
