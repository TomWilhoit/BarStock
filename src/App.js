import React, { Component } from 'react';
import MockData from './mockData.js'
import './css/App.css';
import Inventory from './Inventory.js';
import Totals from './Totals.js';
import Header from './Header.js';
import Footer from './Footer.js';
import Login from './Login.js';



class App extends Component {
  constructor() {
    super();
    this.state = {
      allProducts: [MockData.distributor[0].inventory], 
      allMenu: [MockData.menu],
      cartItems: [],
      totalCost: 0,
      totalProjected: 0,
      currentUser: ""
    }
  }
  toggleAlert() {
    alert('toggle');
  }
  render() {
    return (
      <div className="App">
        {/* <div className="Login-container">
          <Login/>
        </div> */}
        <div className="Header-container">
          <Header/>
        </div>
        <div className="Inventory-container">
          <Inventory allProducts={this.state.allProducts}/>
        </div>
        <div className="Totals-container">
          <Totals allProducts={this.state.allProducts} 
                  allMenu={this.state.allMenu} />
        </div>
        <div className="Footer-container">
          <Footer/>
        </div>
      </div>
    )
  }
}

export default App;
