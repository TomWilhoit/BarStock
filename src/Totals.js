import React, { Component } from 'react';
import MockData from './mockData.js';
import App from './App.js';
import Inventory from './Inventory';
import './css/Totals.css';


class Totals extends Component {
  constructor(props) {
    super(props);
    this.state ={
      cartObjs: props.cartItems, 
      cartNew: props.cartMenu
    }
  }

  

  totalCost = () => {
    return this.state.cartObjs.reduce((acc, currObj) => {
      return acc += currObj.price; 
    }, 0)
  }

  

  totalProfit = () => {
    let matchingObject; 
    return this.state.cartObjs.reduce((acc, currObj) => {
      
      
  
        matchingObject = this.state.cartNew.find(item => {
          return item.inventory_code === currObj.inventory_code;
        });
        
        console.log(matchingObject.product)
      
      
    
        return acc; 
    }, 0)
  }



  
  render() {
    // console.log(this.state)
    return (
      <div className="Totals">
        <h3>Total Cost</h3>
        <h2>{this.totalCost()}</h2>
        <br/>
        <h3>Potential Profit</h3>
        <h2>{this.totalProfit()}</h2>
        {/* <h2>{this.totalProfit()}</h2> */}
      </div>
    );
  }
};






export default Totals;