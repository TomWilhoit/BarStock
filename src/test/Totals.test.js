import React from "react";
import App from "../App";
import { shallow } from "enzyme";
import Totals from "../Totals";

describe("Totals", () => {
  let wrapper;
  let allMenu = [
    {
      product: "Fireball",
      inventory_code: 10001,
      price_per_drink: 3,
      serving_size: 1.5,
      unit: "ounces",
      tier: "mid"
    },
    {
      product: "Jim Beam",
      inventory_code: 10002,
      price_per_drink: 5,
      serving_size: 1.5,
      unit: "ounces",
      tier: "mid"
    }
  ];
  let cartItems = [
    {
      product: "Fireball",
      inventory_code: 10001,
      type: "liquor",
      category: "whiskey",
      price: 18.78,
      size: 33.8,
      unit: "ounces"
    },
    {
      product: "Jim Beam",
      inventory_code: 10002,
      type: "liquor",
      category: "whiskey",
      price: 16.99,
      size: 25.3,
      unit: "ounces"
    }
  ];
  let menuObj = {
    product: "Fireball",
    inventory_code: 10001,
    price_per_drink: 3,
    serving_size: 1.5,
    unit: "ounces",
    tier: "mid"
  };
  let changeCart = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Totals
        cartMenu={allMenu}
        cartItems={cartItems}
        changeCart={changeCart}
      />
    );
  });
  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
