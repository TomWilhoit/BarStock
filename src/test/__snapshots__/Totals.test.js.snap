// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Totals should match snapshot when all data is passed correctly 1`] = `
<article
  className="Totals"
>
  <div
    className="total-label"
  >
    <h3>
      Current Order
    </h3>
  </div>
  <section
    className="cart-display"
  >
    <section
      className="cart-products"
      key="0"
    >
      <div
        className="item-product"
      >
        <h5>
          1 X 
          Fireball
        </h5>
      </div>
      <div
        className="item-price"
      >
        <h5>
          @ $
          18.78
        </h5>
      </div>
      <div
        className="item-delete"
      >
        <button
          onClick={[MockFunction]}
        >
          <i
            className="fas fa-times"
          />
        </button>
      </div>
    </section>
    <section
      className="cart-products"
      key="1"
    >
      <div
        className="item-product"
      >
        <h5>
          1 X 
          Jim Beam
        </h5>
      </div>
      <div
        className="item-price"
      >
        <h5>
          @ $
          16.99
        </h5>
      </div>
      <div
        className="item-delete"
      >
        <button
          onClick={[MockFunction]}
        >
          <i
            className="fas fa-times"
          />
        </button>
      </div>
    </section>
  </section>
  <div
    className="cost-container"
  >
    <section
      className="returns-cost"
    >
      <h2>
        Potential Returns*
      </h2>
      <h3>
        $
        116.16
      </h3>
    </section>
    <section
      className="total-cost"
    >
      <h2>
        Total Cost
      </h2>
      <h3>
        $
        35.77
      </h3>
    </section>
  </div>
</article>
`;
